### PR TITLE
make the snaplist code more async

### PR DESCRIPTION
**Pull request by mwhudson**

### CODE REVIEW

File: subiquity/snapd.py [hunk 1]
```diff
@@ -146,6 +146,7 @@ def post(self, path, body, **args):
             "Don't know how to fake POST response to {}".format((path, args)))
 
     def get(self, path, **args):
+        time.sleep(1)
```
Comment (xnox) [inline subiquity/snapd.py]: Is this ratelimitting talking to snapd? Or some race? Do you want for this to be blocking, or like await sleep?

Comment (xnox) [reply subiquity/snapd.py]: Or is it like for debug "fake" mode, and should be pushed further down?

Comment (mwhudson) [reply subiquity/snapd.py]: This is the fake stuff, and as far down as it can be pushed I think?

Comment (xnox) [reply subiquity/snapd.py]: oh yes.... all of the dirs are fake. Somehow i thought it's constructing a URL.

**Pull request merged**